db/wal_manager.cc

```diff
@@ -322,8 +322,16 @@ Status WalManager::GetSortedWalsOfType(const std::string& path,
             continue;
           }
         }
-      }
-      if (!s.ok()) {
+      } else if (!s.ok()) {
+        if (log_type == kArchivedLogFile &&
+            (s.IsNotFound() || 
+             (s.IsIOError() && env_->FileExists(LogFileName(path, number)).IsNotFound()))) {
```

Comment:
Should it be `ArchivedLogFileName()` rather than `LogFileName()`?

Reply:
Never mind, I think you are right. This function confuses me.

Reply:
It confuses me too :smile:
Let me have another look at it...